project_weather_app/database.py

```diff
@@ -49,8 +86,16 @@ def reset(self):
         self.conn.commit()
         print("Database table 'weather' is deleted.")
 
+    def drop(self):
+        self.cursor.execute(''' 
+            DROP TABLE IF EXISTS weather; 
+        ''')
+        self.conn.commit()
+        print("Database table 'weather' is dropped.")
+
     def __del__(self):
         """
         Close connection when object is deleted.
         """
-        self.conn.close()
+        if self.conn is not None:
+            self.conn.close()
```

Comment:
personally since i am a more defensive coder, i wouldn't give Python the responsibility to close the connection whenever it deletes your variable, but instead have a dedicated `close` class method that you call to close a connection, since you have more control about when the connection should actually be closed